Packages/com.chisel.editor/Chisel/Editor/UtilityWindows/SurfaceEditorWindow/SurfaceEditor.GUI.cs

```diff
@@ -0,0 +1,302 @@
+{
+    internal partial class SurfaceEditor
+    {
+        private float           m_PanIncrementU   = 1.0f;
+        private float           m_PanIncrementV   = 1.0f;
+        private float           m_RotIncrement    = 15.0f;
+        private float           m_ScaleIncrementU = 1.0f;
+        private float           m_ScaleIncrementV = 1.0f;
+        private int             m_CurrentTab      = 0;
+        private LayerUsageFlags m_LayerUsage      = LayerUsageFlags.All;
+
+        // $TODO: find a way to edit surface flags outside of model... its serialized by unity and not visible to SurfaceReference or ChiselBrushMaterial.
+        // ^ this can be done, reference new EditorTools code for this
+        //private SurfaceFlags    m_WorldSpaceTex = SurfaceFlags.TextureIsInWorldSpace;
+
+        private void DrawTabBar()
+        {
+            GUILayout.BeginHorizontal();
+
+            m_CurrentTab = GUILayout.Toolbar
+            (
+                m_CurrentTab, new string[] { "Flags", "UVs", "Custom", "Other" }, m_TabStyle
+            );
+
+            GUILayout.EndHorizontal();
+        }
+
+        private void DrawWindowContents()
+        {
+            switch( m_CurrentTab )
+            {
+                case 0: // Flags
+                {
+                    DrawFlagsTabContent();
+                    break;
+                }
+
+                case 1: // UVs
+                {
+                    DrawUVTabContent();
+                    break;
+                }
+
+                case 2: // Custom
+                {
+                    DrawCustomTabContent();
+                    break;
+                }
+
+                case 3: // Other
+                {
+                    DrawOtherTabContent();
+                    break;
+                }
+            }
+        }
+
+        // Surface flags - visible, receive shadows, collidable, cast shadows, lock texture to object
+        private void DrawFlagsTabContent()
+        {
+            m_LayerUsage = GetFlagsUI( m_LayerUsage );
+
+            if( CurrentSurfaceSelection != null )
+            {
+                foreach( SurfaceReference surfaceReference in CurrentSurfaceSelection )
+                {
+                    ChiselBrushMaterial bm = surfaceReference.BrushMaterial;
+
+                    if( bm.LayerUsage != m_LayerUsage )
+                        bm.LayerUsage = m_LayerUsage;
+                }
+            }
+        }
+
+        // UV Editing - translation, scale, rotation, anchor/pivot, auto fit
+        private void DrawUVTabContent()
+        {
+            GUILayout.BeginHorizontal(); // pan/rotation/scale
+            {
+                GUILayout.Space( 4 );
+
+                GUILayout.BeginVertical( "Tooltip", GUILayout.Width( 194 ) ); // Pan/Nudge
+                {
+                    GUILayout.BeginHorizontal();
+                    {
+                        GUILayout.Label( "Pan",       "OL Title", GUILayout.Width( 64 ) );
```

Comment:
FYI: 
GUILayout.Width should be a static variable, since it does an allocation
string to GUIContent and string to GUIStyle are allocations as well
Fine while prototyping, but something to be aware of